src/main/java/Library/Service/BookLibraryService.java

```diff
@@ -82,9 +85,26 @@ public boolean removeBook(int id) {
         return true;
     }
 
+    // @Override
+    // public List<Book> searchBook(String title) {
+    // return bookManager.SearchByName(title);
+    // }
+
+    // @Override
+    // public List<Book> search(SearchStrategy strategy, String keyword) {
+    // List<Book> books;
+    // return strategy.search(keyword, books);
+    // }
+
     @Override
-    public List<Book> searchBook(String title) {
-        return bookManager.SearchByName(title);
+    public List<Book> search(TitleSearchStrategy titleSearchStrategy, String keyword) {
+        List<Book> books = bookManager.SearchByName(keyword);
+        return books;
```

Comment:
Does the strategy pattern being used in here?

Reply:
This is more like using polymorphism to make search work. it should be more like this in my opinion

```java
SearchStrategy searchStrategy= new TitleSearchStrategy();
searchStrategy.search("title");
```